services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/FeatureGateAspect.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.openhouse.tables.toggle;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.openhouse.internal.catalog.toggle.IcebergFeatureGate;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * To use {@link TableFeatureToggle} one can add implementation of feature-toggle checking in this
+ * class with different types of advises injected (e.g. {@link Before}).
+ */
+@Aspect
+@Component
+@Slf4j
+public class FeatureGateAspect {
+
+  @Autowired private TableFeatureToggle tableFeatureToggle;
+
+  @Before("@annotation(featureGate)")
+  public void checkIcebergFeatureFlag(JoinPoint joinPoint, IcebergFeatureGate featureGate) {
+    TableIdentifier tableIdentifier = obtainIdFromAnnotationArgs(joinPoint);
```

Comment:
`TableIdentifier` makes me think its better to leave the class in internalcatalog, but rest of the classes look good in tables

Reply:
This feature aspect is not going to only server internal catalog though. 
My intention for it is a central annotation behavior's registration hub. For now, we have an annotation in catalog but we will soon add a new one which expects `TableDto` as part of signature. 

The perfect solution will be to have a cluster:annotation module for such central annotation registration.